### PR TITLE
[FIX] l10n_fr_pos_cert: Remove statement protection

### PR DESCRIPTION
**Pull request by GaetanVandenBergh**
State field no more exist
Protection on statement line is enough
if you cannot change any statement line,
it is not important how you group them in statements

opw-3225322

Description of the issue/feature this PR addresses:
State field no more exist in model account.bank.statement
Checked with @poma-odoo 
Protection on statement line is enough
if you cannot change any statement line, it is not important how you group them in statements

Current behavior before PR:
Trying to delete a statement will generate a traceback
`AttributeError: 'account.bank.statement' object has no attribute 'state'`

Desired behavior after PR is merged:
No traceback and deletion works as expected

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by GaetanVandenBergh** [comment]
Duplicate of https://github.com/odoo/odoo/pull/114637

File: addons/l10n_fr_pos_cert/models/account_bank_statement.py [hunk 1]
```diff
@@ -1,31 +0,0 @@
...
-    def _unlink_except_created_by_pos(self):
-        for statement in self:
-            if not statement.company_id._is_accounting_unalterable() or not statement.journal_id.pos_payment_method_ids:
-                continue
-            if statement.state != 'open':
-                raise UserError(_('You cannot modify anything on a bank statement (name: %s) that was created by point of sale operations.') % statement.name)
-
-
-
-class AccountBankStatementLine(models.Model):
```
Comment (poma-odoo) [inline addons/l10n_fr_pos_cert/models/account_bank_statement.py:21]: We need to check statement lines and prevent them from changing, we just don't need the statements themselves.
So this override should stay and only the above one should be removed

Comment (GaetanVandenBergh) [reply addons/l10n_fr_pos_cert/models/account_bank_statement.py:21]: Statement line check the statement state too
A PR already exist for this issue, I have closed this one as it's just a duplicate

Comment (poma-odoo) [reply addons/l10n_fr_pos_cert/models/account_bank_statement.py:21]: Now, statement line is independent of the statement, so you can access journal_id and company_id from the line itself. There is no state field anymore, so the line should never be modified or unlinked in this case.

**Pull request closed without merging**